[T16-2] SuperTA

yuheong opened the pull request:
@wenshei @keithtan @Caephler @darieca

tanhengyeow commented on docs/UserGuide.adoc:
```diff
@@ -12,19 +12,19 @@ ifdef::env-github[]
+.  Copy the file to the folder you want to use as the home folder for your SuperTA client.
 .  Double-click the file to start the app. The GUI should appear in a few seconds.
 +
 image::Ui.png[width="790"]
```
Might want to consider updating the picture to your mockup of the app.

tanhengyeow reviewed (commented):
Good effort in updating the required documents! Great to see the comments made in tutorial reflected in the documents :) Please remember to update `README.adoc` page as per the requirements.

Please work towards the v1.1 milestone and update the documents according to any updates. Also, please do not close this PR as feedback would be made in this PR for subsequent milestones 😄

tanhengyeow reviewed (commented):
Good job for v1.1! Some points to note:
1. Please acknowledge AB4 under acknowledgements, something like `AddressBook-Level4 project created by SE-EDU initiative at https://github.com/se-edu/`
2. Include features that are `Not yet implemented` as `Coming in v2.0`.

💪 💪 for the next milestone :)

tanhengyeow reviewed (commented):
Good work so far! As mentioned in tutorial, do prioritize the remaining tasks and break them down into manageable chunks for the rest of the milestones 😄 

Ideally, all the major tasks should be done by v1.3 so the team can focus on fixing bugs, docs and handling minor updates after that.

Caephler committed:
Update EOF

Caephler committed:
Partial checkstyle fix

Caephler committed:
Fix tests and style

Caephler committed:
Fix error not thrown

Caephler committed:
Merge branch 'enhancement-docs-fixes-tests' of https://github.com/Caephler/main into enhancement-docs-fixes-tests

darieca committed:
model API now includes update assignment

darieca committed:
create UpdateAssignmentCommand and parsers

Commit:
Mark-attendance correctly marks students indicated as present within a session

Commit:
List-tutorial-groups now outputs attendance sessions for each tutorial group

yuheong committed:
Merge branch 'master' into create-attendance

Commit:
Resolve checkstyle errors in ModelManager

Commit:
Rename contains(StudentId) to containsId

Caephler committed:
Merge pull request #97 from triger15/create-attendance

Create and mark attendance commands

Caephler committed:
Merge branch 'master' into enhancement-docs-fixes-tests

Caephler committed:
Merge pull request #101 from Caephler/enhancement-docs-fixes-tests

Enhancement docs fixes tests

Caephler committed:
Fix breaks and ready for release

Caephler committed:
Merge pull request #102 from Caephler/v1.3release

Fix breaks and ready for release

Caephler committed:
Update README.adoc

darieca committed:
[UpdateAssignmentCommand] add in modified max mark field

darieca committed:
[UpdateAssignmentCommandParser] parse UpdateAssignmentCommand

darieca committed:
update models to accept UpdateAssignment API

darieca committed:
add test for UpdateAssignmentComamnd and parser

darieca committed:
update modelstub other tests

Caephler committed:
Add initial PPP

Caephler committed:
Draft PPP

wenshei committed:
changed AddressBook references to SuperTA

wenshei committed:
Removed Appendix: Product Survey

wenshei committed:
changed addressbook references to superta

wenshei committed:
changed addressbook references to superta

wenshei committed:
Merge pull request #100 from wenshei/update_DeveloperGuide

Updated DeveloperGuide to match SuperTA

darieca committed:
add change assignment title field in UpdateAssignmentCommand.java

Caephler committed:
Add initial view for Session

darieca committed:
update find command UG

darieca committed:
update find command UG

Caephler committed:
Fix some bugs

Caephler committed:
Update docs

Caephler committed:
Update version

Caephler committed:
Update line endings

Caephler committed:
Fix style errors

Caephler committed:
Merge pull request #106 from Caephler/enhancement-ui-attendance

Enhancement ui attendance

darieca committed:
Merge branch 'darieca-darieca-docs-update-UG'

darieca committed:
Merge remote-tracking branch 'upstream/master' into darieca-docs-update-UG

darieca committed:
Merge pull request #107 from darieca/darieca-docs-update-UG

update UG find command

keithtan committed:
Update User Guide

yuheong committed:
Add tests for feedback and create attendance command

yuheong committed:
Add tests for session

Commit:
Add tests for attendance

Commit:
Add tests for mark attendance command

Commit:
Edit invalid messages for Mark Attendance Command

Commit:
Fix checkstyle errors

Commit:
Add storage tests for attendance and session

Commit:
Fix checkstyle erros and add minor test for session

Caephler committed:
Merge pull request #110 from keithtan/UpdateUserGuide

Update User Guide

Commit:
Add minor parser test

Commit:
Parser now accepts lowercase Student ID

ParserUtil.java will now use toUpperCase() method on all student IDs
that have been entered in the input. Implemented tests in
ParserUtilTest.java as well.

Commit:
Fix checkstyle errors

keithtan committed:
Update example commands

Commit:
Add toUpperCase method to StudentId class

keithtan committed:
Find command accepts multiple keywords

keithtan committed:
Update UserGuide

keithtan committed:
Add tests for FindCommandParser

keithtan committed:
Update delete command example

keithtan committed:
Update delete command description

Caephler committed:
Merge pull request #134 from keithtan/updateUserGuide

Update User Guide

Caephler committed:
Merge pull request #131 from triger15/attendance-tests

Added tests for create attendance command, mark attendance command

Caephler committed:
Merge pull request #133 from triger15/case-insensitive

Parser now accepts lowercase Student ID

Caephler committed:
Update StateEvent to trigger from each event that modified the underlying model.

Caephler committed:
Fix style errors

Caephler committed:
Fix - no longer able to add student with same ID to client

Caephler committed:
Fix breaking tests and a fix for EditCommand

Caephler committed:
Merge pull request #135 from Caephler/fix-ui-refresh

Fix UI Refresh and Add / Edit Command

keithtan committed:
Attempt tests for FindCommandParser

yuheong committed:
Update reposense

Reposense now correctly shows commit info for all authors

Caephler committed:
Merge pull request #136 from keithtan/UpdateFindCommand

Update find command

Caephler committed:
Add view-session command

Caephler committed:
Update user guide for view-session

Caephler committed:
Update user guide with fix

Caephler committed:
Add click event for session detail and add absentee list

Caephler committed:
Add render deleted method

Caephler committed:
Add authorings

Caephler committed:
Update add-to-tutorial-group documentation

Caephler committed:
Edit config

Caephler committed:
Add tests for ViewSessionCommand

Caephler committed:
Add view-session tests

Caephler committed:
Fix XML data path problems and some UI changes

Caephler committed:
Add seed file when starting with no data

Caephler committed:
Edit clear UG description

Caephler committed:
Fix tests and imports

Caephler committed:
Add config.json

Caephler committed:
Change line endings

Caephler committed:
Update reposense config to ignore major refactoring commits

Caephler committed:
Merge pull request #138 from Caephler/enhancement-general-fixes

Enhancement general fixes

Caephler committed:
Initial draft of PPP for Gabriel

Caephler committed:
Merge branch 'docs-gabriel-ppp' of https://github.com/Caephler/main into docs-gabriel-ppp

Caephler committed:
Update documentation and include PPP

Caephler committed:
Update README and UI images

Caephler committed:
Update typo

Caephler committed:
Fix assignment UI not refreshing sometimes

wenshei committed:
fixed empty feedback being stored as feedback

wenshei committed:
added draft PPP, added tags in UG

wenshei committed:
modified PPP

wenshei committed:
Modified DG diagrams to include Attendance in the Model

wenshei committed:
added DeleteAssignment feature implementation in DG

keithtan committed:
Update PPP

keithtan committed:
added help message for duplicate marking

keithtan committed:
added help message for duplicate student names

keithtan committed:
Refactor help messages

Caephler committed:
Merge pull request #142 from wenshei/fix_emptyFeedback

Fixed empty feedback being stored as feedback

Caephler committed:
Merge pull request #141 from Caephler/fix-assignment-ui

Fix assignment UI not refreshing sometimes

Caephler committed:
Merge pull request #139 from Caephler/docs-gabriel-ppp

Documentation: UGDG updates and Personal Portfolio Page

keithtan committed:
Amend checkstyle and add test

keithtan committed:
Merge branch 'master' into UpdatePPP

keithtan committed:
Update PPP

keithtan committed:
Edit help message for GradeAssignmentCommand

Caephler committed:
Fix session

Caephler committed:
Merge pull request #143 from keithtan/UpdateHelpMessages

Update help messages

Caephler committed:
Update PPP

Caephler committed:
Update version number and fix seed file

Caephler committed:
Fix set equality issue

Caephler committed:
Remove unused imports

Caephler committed:
Fix breaking test

Caephler committed:
Merge pull request #144 from Caephler/fix-session

Fix session (v1.3.2 release)

Caephler committed:
Fix incorrect session working

Caephler committed:
Update version number

Caephler committed:
Remove print statements

Caephler committed:
Merge pull request #145 from Caephler/fix-attendance

Fix attendance (v1.3.3)

darieca committed:
[UpdateAssignmentCommand] add UpdateAssignmentDescriptor, and change to optional title/maxmark field

darieca committed:
add new syntax for new assignment title / marks

darieca committed:
parser now accepts UpdateAssignmentDescriptor

darieca committed:
[Model] update UpdateAssignment API

darieca committed:
add exception for duplicate assignment title

darieca committed:
[TutorialGroup.java] add updateAssignment method

darieca committed:
add placeholders for tests

wenshei committed:
updated PPP and AboutUs

keithtan committed:
Update UI component class diagram

darieca committed:
Merge branch 'master' into update-assignment-command

keithtan committed:
Update Appendix F

keithtan committed:
Add Delete Student Use case

Caephler committed:
Merge pull request #151 from keithtan/UpdateDeveloperGuide

Update developer guide

darieca committed:
[UpdateAssignmentCommand] minor ahjustment

darieca committed:
*.java

add parser test

darieca committed:
Merge remote-tracking branch 'upstream/master' into update-assignment-command

darieca committed:
add parser test

darieca committed:
[UpdateAssignmentCommand] update syntax

darieca committed:
fix checkstyle errors

darieca committed:
fix checkstyle errors

darieca committed:
fix checkstyle errors

darieca committed:
update/delete UpdateAssignment tests

darieca committed:
update UpdateAssignment tests

darieca committed:
format fixes

darieca committed:
[userguide] include update-assignment details and exceptions + formatting fix

Caephler committed:
Merge pull request #152 from darieca/update-assignment-command

Update assignment command

keithtan committed:
Merge branch 'master' into UpdatePPP

keithtan committed:
Add PPP

keithtan committed:
Update PPP

keithtan committed:
Add new line at EOF

Caephler committed:
Update PPP and fix seed file issue

Caephler committed:
Merge pull request #153 from keithtan/UpdatePPP

Add PPP

wenshei committed:
modified DG

wenshei committed:
Merge branch 'master' into update_PPP

wenshei committed:
Merge pull request #155 from wenshei/update_PPP

Add PPP and updated DeveloperGuide

wenshei committed:
directed portfolio links in AboutUs to individual PPP

wenshei committed:
modified DG

wenshei committed:
Added use cases for grade assignment, mark attendance, give feedback

wenshei committed:
added tests for AddStudentToTutorialGroupCommandParser, and equals method for command

wenshei committed:
added equals method for CreateAssignmentCommand

wenshei committed:
Added CreateAssignemntCommandParserTest

wenshei committed:
Added equals method for RemoveStudentFromTutorialGroupCommand

wenshei committed:
Added RemoveStudentFromTutorialGroupCommandParserTest

wenshei committed:
Added equals method for ViewTutorialGroupCommand

wenshei committed:
Added ViewTutorialGroupCommandParserTest

wenshei committed:
removed unused imports for test

wenshei committed:
Added CreateTutorialGroupCommandTest

wenshei committed:
Modified DeleteTutorialGroupCommandTest to include positive test case

wenshei committed:
Added equals method for Grade

wenshei committed:
Added equals method for GradeAssignmentCommand

wenshei committed:
modified for checkstyle

wenshei committed:
modified for checkstyle

yuheong committed:
Merge pull request #154 from Caephler/docs-gabriel-ppp-final

Update PPP and fix seed file issue

yuheong committed:
Merge pull request #157 from wenshei/add_tests

Added tests for commands and parsers

yuheong committed:
Update DeveloperGuide.adoc

darieca committed:
add PPP

darieca committed:
updateUG

Caephler committed:
Merge pull request #156 from wenshei/update_DeveloperGuide

Updated developer guide

keithtan committed:
Update User Guide

keithtan committed:
Insert new line after parameters

keithtan committed:
Update User Guide

keithtan committed:
Make some small changes

darieca committed:
Merge branch 'master' into docs-PPP-DG

darieca committed:
[DG] include find and update-assignment implementation

darieca committed:
edit PPP

darieca committed:
[DG] include usecase and manual testing for find and update assignment command

darieca committed:
image for DG

darieca committed:
edit PPP

darieca committed:
Merge pull request #158 from keithtan/UpdateUserGuide

Update user guide

darieca committed:
Merge branch 'master' into docs-PPP-DG

darieca committed:
update DG

darieca committed:
fix checkstyle

darieca committed:
fix checkstyle

darieca committed:
fix checkstyle

Commit:
Update DG with MarkAttendanceCommand implementation

Commit:
Fix existing errors present in DG

Commit:
Edit class diagram and minor tweaks for setting PPP

Commit:
Update PPP

darieca committed:
update test

darieca committed:
update diagram

darieca committed:
fix typo

darieca committed:
delete wrong file

Caephler committed:
Update TutorialGroupDesign.adoc

Caephler committed:
Merge pull request #160 from triger15/update-dg

Update DG and PPP

darieca committed:
file restored

darieca committed:
Merge branch 'master' into docs-PPP-DG

Caephler committed:
Merge pull request #159 from darieca/docs-PPP-DG

Add PPP and update DG

wenshei committed:
updated PPP

wenshei committed:
updated PPP

darieca committed:
update test

Caephler committed:
Update UG and remove LOs from DG

Caephler committed:
Merge pull request #161 from Caephler/docs-update-ug-screenshots

Update UG and remove LOs from DG

wenshei committed:
modified various Command messages for formatting

darieca committed:
update parser tests

darieca committed:
remove original max marks field

darieca committed:
match updated info to UG

Caephler committed:
Update UGDG

darieca committed:
update updateassignment test to work with current implementation

darieca committed:
Merge branch 'master' into various-tests

darieca committed:
edit PPP

Caephler committed:
Merge pull request #162 from Caephler/update-ugdg-ppp

Update UGDG and PPP

darieca committed:
update DG for update assignment manual testing

darieca committed:
Merge branch 'master' into various-tests

darieca committed:
fix tests

darieca committed:
fix checkstyle

wenshei committed:
updated PPP

wenshei committed:
modified undo/redo diagrams in DG

Commit:
View feedback takes in lowercase student-id

Commit:
Minor typo in XmlAdaptedFeedback

Commit:
Increase test coverage for viewing multiple feedback

yuheong committed:
Merge pull request #166 from wenshei/update_PPP

Updated PPP and DG

Caephler committed:
Merge pull request #163 from darieca/various-tests

update-assignment fix + SuperTaClientParserTests

Caephler committed:
Merge pull request #167 from triger15/viewfeedback-bug

View feedback bug, increase test coverage for view-feedback

Caephler committed:
Merge branch 'master' into update_command_messages

Caephler committed:
Merge pull request #164 from wenshei/update_command_messages

Modified various Command messages for formatting

Commit:
Update ug, dg, ppp

Caephler committed:
Merge pull request #168 from triger15/update-docs

Update ug, dg, ppp

Caephler committed:
Update UserGuide

Caephler committed:
Update version

Caephler committed:
Fix feedback truncation

Commit:
Final updates to fix PPP not rendering properly

Commit:
Minor typos

Caephler committed:
Merge pull request #170 from triger15/last-updates

Final updates to fix PPP not rendering properly

Caephler committed:
Merge pull request #169 from Caephler/v1.4release

v1.4 Release
